src/ContractCallDataEncoder.cpp

```diff
@@ -361,7 +361,7 @@ QVariant ContractCallDataEncoder::decode(SolidityType const& _type, bytes const&
 	else if (type == QSolidityType::Type::Bool)
 		return QVariant::fromValue(toString(decodeBool(rawParam)));
 	else if (type == QSolidityType::Type::Bytes || type == QSolidityType::Type::Hash)
-		return QVariant::fromValue(toChar(decodeBytes(rawParam)));
+		return QVariant::fromValue(QString::fromStdString(toHex(decodeBytes(rawParam), 2, HexPrefix::Add)));
 	else if (type == QSolidityType::Type::String)
 		return QVariant::fromValue(toChar(decodeBytes(rawParam)));
```

Comment:
now we convert to the corresponding hex value in case of bytes type 
and to ascii in case of string type.
before this change we tried to convert to ascii even in case of bytes type.

It could lead to a problem if the bytes type actually contains ascii readable content. User will expect to see ascii char and not hex value. so we could add an action to switch between hex and ascii conversion.
This is kind of the same story as https://github.com/ethereum/mix/issues/300